### ISSUE TITLE
Embedded nodes

### ISSUE THREAD

**Issue by aloisdeniel** [superlistapp/super_editor#1581]
Hey, 👋

I'm trying to add support for embedded nodes (_like list items, code blocks, or images_) inside other stylized blocks (_like blockquote_).

Do the editor support nodes inside nodes?

Thanks,

**Comment by matthew-carroll** [comment]
There's no built-in support for that. If you're going to make that work, you'd need to take your sub-nodes and then wrap them in a single overall node, which then delegates its API to those sub-nodes.

You might want to check whether that's truly something you need to do. None of our clients over the past few years have ever needed that feature. I don't know what you're trying to build, because there wasn't any mention in your post, but you might be able to reframe the problem and avoid the need for a deeper document tree.

**Comment by aloisdeniel** [comment]
I work for ClickUp and we're trying to evaluate super_editor for implementing its document editor. 

We have a few advanced use cases like multiple columns of contents :

<img width="782" alt="image" src="https://github.com/superlistapp/super_editor/assets/7687231/4b819613-e376-418f-9797-c92627b642ec">

Appart from that I guess we could find another way to implement embedded nodes by cumulating multiples styles on each node and styling consecutive nodes with the same styles to create a container around them.

**Comment by matthew-carroll** [comment]
It looks like your goal is a block-style editor, similar to Notion's layout, or some other popular modern apps. Is that right?

I always figured we'd eventually make block editing possible. I think that's mostly an issue of funding. Our existing funding clients (Superlist, Clearful, Turtle, Bringing Fire, Reflection) only utilize the more traditional single-column document layout. Expanding that support to arbitrary blocks would be large enough as to require funding for that effort.

Would ClickUp consider funding, or partially funding development of the relevant features to bring arbitrary block editing?

**Comment by aloisdeniel** [comment]
Yes, we're aiming for something pretty similar to other block-based editors like Notion.

Since we're still in an evaluation phase, I can't guarantee any funding, but I will definitely share the idea internally!

**Comment by matthew-carroll** [comment]
Hi @aloisdeniel did you get a chance to follow up on the possibility of ClickUp helping to fund block layout for Super Editor?

**Comment by aloisdeniel** [comment]
Hey! 

We're still in discussion with the management, but funding is an option we're considering!

I'll keep you posted!
